### DISCUSSION TITLE
Velocity marker scores for each cluster

### DISCUSSION THREAD

**Discussion by maddemaddi** [theislab/scvelo#734] [Q&A]
Hi,

Thanks for providing this super helpful tool, we have been able to use it without any problems so far! I have one question regarding the velocity scores calculated by scv.tl.rank_velocity_genes() and scv.tl.rank_dynamical_genes() - I am able to extract velocity scores for each gene by calling adata.var and this gives me one table for the whole dataset. I can however not figure out how to extract the velocity scores for each specific cluster individually, could you please let me know how to do this :)

Br,

Madeleine

**Comment by WeilerP** [comment]
@maddemaddi, sorry I'm a bit confused: velocity scores are a property of genes (hence stored in `adata.var`), clusters of cells (usually stored in `adata.obs`). Could you please clarify what you mean by velocity scores for each specific cluster?

**Reply by maddemaddi** [reply]
Hi,

Thanks for the quick reply. The scv.tl.rank_velocity_genes() function gives a ranking of velocity genes in a table which ranks the genes separately for each cell cluster, like this:

<img width="706" alt="Screenshot 2021-10-21 at 11 27 33" src="https://user-images.githubusercontent.com/74403229/138240566-dfb4e1ac-ddde-4903-80be-6f136f795955.png">

What I am trying to ask is, how do I pull out numerical values that are laying the ground for these cluster-specific rankings?

Thanks!
